metron-platform/metron-common/src/main/java/org/apache/metron/common/bolt/ConfiguredEnrichmentBolt.java

```diff
@@ -17,18 +17,41 @@
  */
 package org.apache.metron.common.bolt;
 
+import java.io.IOException;
 import java.lang.invoke.MethodHandles;
+import java.util.Map;
+
+import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.metron.common.configuration.EnrichmentConfigurations;
+import org.apache.metron.stellar.common.utils.HttpClientUtils;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public abstract class ConfiguredEnrichmentBolt extends ConfiguredBolt<EnrichmentConfigurations> {
 
   private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
+  protected CloseableHttpClient httpClient;
 
   public ConfiguredEnrichmentBolt(String zookeeperUrl) {
     super(zookeeperUrl, "ENRICHMENT");
   }
 
+  @Override
+  public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+    super.prepare(stormConf, context, collector);
```

Comment:
Do we necessarily want this in all cases? What about users that never need or want this function?

Reply:
I can add a property to configure this.  Should it be included by default or not ?  

Do you think instantiating this object but never using it would be harmful?  It might make things simpler for users if there is one less setting to worry about.

Reply:
More specifically, shouldn't this be isolated to the stellar function itself? This seems like a bleeding of concerns. With zookeeper, it makes sense to me because we're taking the architectural position that enabling dynamic, real-time configuration loading is part of the framework. I don't think that same general applicability applies to HttpClients. What about using the initialize method instead?

```
public interface StellarFunction {
  Object apply(List<Object> args, Context context) throws ParseException;
  void initialize(Context context);
  boolean isInitialized();
}
```

Reply:
I tried that initially but realized there is no shutdown hook in Stellar.  HttpClient objects need to be closed and the only hook that exists is in the bolts.  

If we did have a way to close it, I would be in favor of managing it in the function.  If we decide it's worth adding that hook in Stellar (I think we should as a follow on) we can easily move this inside the function.

Reply:
I'm inclined to add the shutdown hook.  This stuck out to me also as bleeding concerns a bit.

Reply:
Adding a shutdown hook in Stellar is not trivial and out of scope for this PR in my opinion.  I would prefer that be a follow on.

Reply:
Actually I am wrong here?  Would adding a shutdown hook be simple?